java/ql/src/experimental/Security/CWE/CWE-555/context.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Context>
+    <!-- BAD: Password of datasource is in not encrypted -->
+    <Resource name="jdbc/exampleDS" auth="Container" type="javax.sql.DataSource"
+               maxTotal="100" maxIdle="30" maxWaitMillis="10000"
+               username="root" password="1234"
+               driverClassName="com.mysql.jdbc.Driver"
+               url="jdbc:mysql://www.example.com:3306/proj"/>
+
+    <!-- GOOD: Password is encrypted and stored in a password vault -->
+    <Resource name="jdbc/exampleDS" auth="Container" type="javax.sql.DataSource"
+               maxTotal="100" maxIdle="30" maxWaitMillis="10000"
+               username="root" password="${VAULT::exampleDS::password::N2NhZDYzOTMtNWE0OS00ZGQ0LWE4MmEtMWNlMDMyNDdmNmI2TElORV9CUkVBS3ZhdWx0}"
```

Comment:
Question to maintainers: Is the "Code scanning" action not working correctly? It shows a lot of false positives:
![Query warning screenshot](https://user-images.githubusercontent.com/11685886/102001106-6a67c280-3cee-11eb-9870-9894f0133281.png)

Reply:
I don't know how the code scanning action works either. 

I didn't find the query to detect password in configuration files within the Java codeql repository while I noticed there are explicit queries for .NET and JavaScript in codeql repositories so I thought a query for Java would be helpful. 

However, this query does help to detect false positives like encrypted passwords and password placeholders while these scenarios are common in Java EE deployment. This query can be further fine tuned for Java EE specific requirements.

Reply:
After a second thought, I think the code scanning action is using the query in the .NET repository to scan this PR since there is another false positive complaining <code>Generic catch clause</code> with <code>csharp/extractor/Semmle.Extraction.CIL/Entities/Method.cs</code> so it seems the tool falsely processed this Java query as a .NET query.

Reply:
Ah yes that makes sense, it looks like this is indeed the C# query [PasswordInConfigurationFile.ql](https://github.com/github/codeql/blob/main/csharp/ql/src/Configuration/PasswordInConfigurationFile.ql).